expr/func_groupbytags.go

```diff
@@ -0,0 +1,146 @@
+package expr
+
+import (
+	"bytes"
+	"errors"
+	"sort"
+	"strings"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncGroupByTags struct {
+	in         GraphiteFunc
+	aggregator string
+	tags       []string
+}
+
+func NewGroupByTags() GraphiteFunc {
+	return &FuncGroupByTags{}
+}
+
+func (s *FuncGroupByTags) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgString{val: &s.aggregator},
+		ArgStrings{val: &s.tags},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncGroupByTags) Context(context Context) Context {
+	// set this? context.consol = consolidation.FromConsolidateBy(s.aggregator)
```

Comment:
I think this should be answered before merging

Reply:
Agreed, but I'm not quite sure how to tell. I think the answer is 'no' after looking through graphite-web code.